pilot/pkg/xds/eds.go

```diff
@@ -418,7 +419,8 @@ func (eds *EdsGenerator) Generate(proxy *model.Proxy, push *model.PushContext, w
 				Resource: util.MessageToAny(l),
 			}
 			resources = append(resources, resource)
-			eds.Server.Cache.Add(builder, token, resource)
+			token := parsePushVersion(push.PushVersion)
```

Comment:
push version does not contain the full state of the world. Endpoints can change without it changing

Reply:
you reminded me, that's a little trouble

Reply:
We can use the PushRequest.Start as the token, it is always increasing